### PR TITLE
Categories statistics

### LINKED ISSUE #11: Statistics of the categorised expenses
# Card:

**Role:** User
**Goal:** Overview about the different expenses
**Reason:** The user wants to see the different types of expenses

---

# Confirmation:

## Acceptance tests:

* pie chart displays percentage of expenses per category
* non-categorized expenses are shown as a category "uncategorized"

### LINKED ISSUE #12: Statistics of the categorised incomes
# Card:

**Role:** User
**Goal:** Overview about the sources of income
**Reason:** The user wants to see his sources of income

---

# Confirmation:

## Acceptance tests:

* A pie chart displays percentage of income sources per category
* Non categorised sources are marked as uncategorised

### PR DESCRIPTION
**Pull request by elimei1**
For incomes (mad) and expenses (wfp1)
Code is from: https://medium.com/@developerchunk/create-custom-pie-chart-with-animations-in-jetpack-compose-android-studio-kotlin-49cf95ef321e

Also done:
Select Frequency in the center
Default date for normal transactions

Closes #12 
Closes #11

### CODE REVIEW

File: app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+                    fontWeight = FontWeight.Medium,
+                    fontSize = 22.sp,
+                    color = Color.Black
+                )
+                Text(
+                    modifier = Modifier.padding(start = 15.dp),
+                    text = data.second.toString(),
+                    fontWeight = FontWeight.Medium,
+                    fontSize = 22.sp,
+                    color = Color.Gray
```
Comment (michael-mezgo) [inline app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt:283]:
```suggestion
                    color = MaterialTheme.colorScheme.secondary
```

File: app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                    .size(height)
+            )
+
+            Column(modifier = Modifier.fillMaxWidth()) {
+                Text(
+                    modifier = Modifier.padding(start = 15.dp),
+                    text = data.first,
+                    fontWeight = FontWeight.Medium,
+                    fontSize = 22.sp,
+                    color = Color.Black
```
Comment (michael-mezgo) [inline app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt:276]:
```suggestion
                    color = MaterialTheme.colorScheme.primary
```

File: app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import at.ac.fhcampuswien.budget_fox.models.Category
+import at.ac.fhcampuswien.budget_fox.models.Transaction
+import at.ac.fhcampuswien.budget_fox.view_models.CategoryViewModel
+import at.ac.fhcampuswien.budget_fox.view_models.TransactionListViewModel
+import at.ac.fhcampuswien.budget_fox.view_models.ViewModelFactory
+import at.ac.fhcampuswien.budget_fox.widgets.SimpleEventIcon
+import at.ac.fhcampuswien.budget_fox.widgets.SimpleTopAppBar
+
```
Comment (michael-mezgo) [inline app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt:52]:
```suggestion
import androidx.compose.material3.MaterialTheme

```

**Review by michael-mezgo — CHANGES_REQUESTED** [review]
Great work 🔝 
I'ld recommend to change the colors, that the text is visible in dark mode

File: app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt (hunk 1 above)
Comment (elimei1) [reply app/src/main/java/at/ac/fhcampuswien/budget_fox/screens/CategoriesStatisticsScreen.kt:283]: Ah yeah everytime

**Review by michael-mezgo — APPROVED** [review]
🔝 🥇

**Pull request merged**